beacon-chain/slasher/chunks.go

```diff
@@ -0,0 +1,553 @@
+//                         |          |          |
+//  max_spans_val_i = [[-, -, -], [-, -, -], [-, -, -]]
+//
+// If C = chunkSize is 3 epochs per chunk, and we input start epoch of chunk 1 which is 3. The next start
+// epoch is the last epoch of chunk 0, which is epoch 2. This is computed as:
+//
+//  last_epoch(chunkIndex(startEpoch)-1)
+//  last_epoch(chunkIndex(3) - 1)
+//  last_epoch(1 - 1)
+//  last_epoch(0)
+//  2
+func (m *MinSpanChunksSlice) NextChunkStartEpoch(startEpoch types.Epoch) types.Epoch {
+	prevChunkIdx := m.params.chunkIndex(startEpoch)
+	if prevChunkIdx > 0 {
+		prevChunkIdx--
+	}
+	return m.params.lastEpoch(prevChunkIdx)
+}
+
+// NextChunkStartEpoch given an epoch, determines the start epoch of the next chunk. For max
+// span chunks, this will be the start epoch of chunk index = (current chunk + 1). For example:
+//
+//                       chunk0     chunk1     chunk2
+//                         |          |          |
+//  max_spans_val_i = [[-, -, -], [-, -, -], [-, -, -]]
+//
+// If C = chunkSize is 3 epochs per chunk, and we input start epoch of chunk 1 which is 3. The next start
+// epoch is the start epoch of chunk 2, which is epoch 6. This is computed as:
+//
+//  first_epoch(chunkIndex(startEpoch)+1)
+//  first_epoch(chunkIndex(3)+1)
+//  first_epoch(1 + 1)
+//  first_epoch(2)
+//  6
+func (m *MaxSpanChunksSlice) NextChunkStartEpoch(startEpoch types.Epoch) types.Epoch {
+	return m.params.firstEpoch(m.params.chunkIndex(startEpoch) + 1)
+}
+
+// Given a validator index and epoch, retrieves the target epoch at its specific
+// index for the validator index + epoch pair in a min/max span chunk.
+func chunkDataAtEpoch(
+	params *Parameters, chunk []uint16, validatorIdx types.ValidatorIndex, epoch types.Epoch,
+) (types.Epoch, error) {
+	requiredLen := params.chunkSize * params.validatorChunkSize
+	if uint64(len(chunk)) != requiredLen {
+		return 0, fmt.Errorf("chunk has wrong length, %d, expected %d", len(chunk), requiredLen)
+	}
+	cellIdx := params.cellIndex(validatorIdx, epoch)
+	if cellIdx >= uint64(len(chunk)) {
+		return 0, fmt.Errorf("cell index %d out of bounds (len(chunk) = %d)", cellIdx, len(chunk))
+	}
+	distance := chunk[cellIdx]
+	return epoch.Add(uint64(distance)), nil
+}
+
+// Updates the value at a specific index in a chunk for a validator index + epoch
+// pair given a target epoch. Recall that for min spans, each element in a chunk
+// is the minimum distance between the a given epoch, e, and all attestation target epochs
+// a validator has created where att.source.epoch > e.
+func setChunkDataAtEpoch(
+	params *Parameters,
+	chunk []uint16,
+	validatorIdx types.ValidatorIndex,
+	epochInChunk,
+	targetEpoch types.Epoch,
+) error {
+	distance := epochDistance(targetEpoch, epochInChunk)
+	return setChunkRawDistance(params, chunk, validatorIdx, epochInChunk, distance)
+}
+
+// Updates the value at a specific index in a chunk for a validator index + epoch
+// pair to a specified, raw distance value.
```

Comment:
There doesn't seem to be any epoch *pair* here, at least not in the parameters list. Maybe something changed?